src/core.cls.php

```diff
@@ -538,6 +538,22 @@ public function send_headers_force($buffer)
 
 		// Hook to modify buffer after
 		$buffer = apply_filters('litespeed_buffer_after', $buffer);
+		
+		// Make sure ESI links are decoded in situation where links are changed by block editor, in render function with wptexturize. Ticket #485348.
+		if(preg_match_all('/(<esi.*src=["\'])(.*)(["\'].*\/>)/mU', $buffer, $matchesEsi)){
+			foreach($matchesEsi[2] as $index => $match){
+				$data = $match;
+				// entity decode needs changes. For example: if finds &#038;
+				if(preg_match('/&#(.*);/mU', $data) === 1){
+					$data = html_entity_decode($data);
+					$data = rawurldecode($data);
+				}
+				
+				if($data !== $match){
+					$buffer = str_replace($matchesEsi[1][$index].$match.$matchesEsi[3][$index], $matchesEsi[1][$index].$data.$matchesEsi[3][$index], $buffer);
```

Comment:
To make it easier to read, set `$matchesEsi[1][$index]` to a var instead of using it twice which makes the row long to read

Reply:
changed